packages/components/src/ListBox/src/ListBoxItem.tsx

```diff
@@ -0,0 +1,313 @@
+
+export const GlobalListBoxItemCssSelector = "hop-ListBoxItem";
+
+export type ListBoxItemSize = "xs" | "sm" | "md" | "lg";
+
+export interface ListBoxItemProps<T> extends StyledComponentProps<Omit<RACListBoxItemProps<T>, "orientation | layout">> {
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size?: ResponsiveProp<ListBoxItemSize>;
+}
+
+interface ListBoxInnerProps extends ListBoxItemRenderProps {
+    /**
+     * The selection indicator to use. Only available if the selection mode is not "none".
+     * When set to "input", the selection indicator will be an either a radio or checkbox based on the selection mode.
+     * @default "check"
+     */
+    selectionIndicator?: "check" | "input";
+    /**
+     * Whether or not the ListBoxItem is in an invalid state.
+     */
+    isInvalid?: boolean;
+    /**
+     * A ListBoxItem can vary in size.
+     * @default "sm"
+     */
+    size: ListBoxItemSize;
+    /**
+     * The children of the ListBoxItem.
+     */
+    children: ReactNode;
+    /**
+     * A function to set the list has selection state.
+     */
+    setListHasSelection: (value: boolean) => void;
+}
+
+const ListBoxItemToIconSizeAdapter = {
+    xs: "sm",
+    sm: "sm",
+    md: "md",
+    lg: "md"
+} as const satisfies SizeAdapter<ListBoxItemSize, IconProps["size"]>;
+
+const ListBoxItemToTextSizeAdapter = {
+    xs: "sm",
+    sm: "sm",
+    md: "sm",
+    lg: "md"
+} as const satisfies SizeAdapter<ListBoxItemSize, TextProps["size"]>;
+
+const getIsListHasSelectionEnd = (isListHasSelection: boolean, prevIsListHasSelectionEnd: boolean): boolean => {
+    return isListHasSelection && !prevIsListHasSelectionEnd;
+};
+
+function ListBoxItemInner(props: ListBoxInnerProps) {
+    const listStateContext = useContext(ListStateContext);
+    
+    const { selectionMode, 
+        isDisabled, 
+        isFocusVisible, 
+        isPressed, 
+        isHovered, 
+        isSelected, 
+        selectionIndicator, 
+        isInvalid, 
+        size, 
+        children,
+        setListHasSelection
+    } = props;
+    
+    const isRadio = selectionIndicator === "input" && selectionMode === "single";
+    const isCheckbox = selectionIndicator === "input" && selectionMode === "multiple";
+    const isCheck = selectionIndicator === "check" && selectionMode !== "none";
+
+    useEffect(() => {
+        setListHasSelection(listStateContext.selectionManager.selectedKeys.size > 0);
+    }, [listStateContext.selectionManager.selectedKeys.size, setListHasSelection]);
+
+    return (
+        <>
```

Comment:
is it possible to add a div here that takes all the available space? i really don't like that we have to setListHasSelection to the parent. It's the kind of properties that should never exists

Reply:
I did have a div before, but I thought we didn't want an extra div.

Reply:
i hate useEffect in components with a passion. 
```
useEffect(() => {
        setListHasSelection(listStateContext.selectionManager.selectedKeys.size > 0);
    }, [listStateContext.selectionManager.selectedKeys.size, setListHasSelection]);
 ```

This makes my skin crawl. So if the price to pay is an extra div, so be it

Reply:
haha ok ya, I know it's not ideal :/ I try to avoid it.. I'll put back the div then.